app/image/color_format.py

```diff
@@ -0,0 +1,77 @@
+"""Support for abstract color formats."""
+
+from enum import Enum
+
+
+class PixelFormat(Enum):
+    """Respresenation defining color hierachy in pixel."""
+
+    RGBA = 1
+    BGRA = 2
+    CUSTOM = 0
+
+
+class Endianness(Enum):
+    """Represenation of color format endianness."""
+
+    LITTLE_ENDIAN = 1
+    BIG_ENDIAN = 2
+
+
+class PixelPlane(Enum):
+    """Representation of color format pixel plane."""
+
+    PACKED = 1
+    PLANAR = 2
+
+
+class ColorFormat():
+    """Representation of color format."""
+    def __init__(self,
+                 pixel_format,
+                 endianness,
+                 pixel_plane,
+                 bpc1,
+                 bpc2,
+                 bpc3,
+                 bpc4=0,
+                 name="unnamed"):
+        self.name = name
+        self.pixel_format = pixel_format
+        self.endianness = endianness
+        self.pixel_plane = pixel_plane
+        self._bpcs = (bpc1, bpc2, bpc3, bpc4)
+
+    @property
+    def bits_per_components(self):
+        return self._bpc
+
+    @bits_per_components.setter
+    def bits_per_components(self, bpcs):
+        try:
+            v1, v2, v3, v4 = bpcs
+        except ValueError:
+            raise ValueError(
+                "Pass an iterable with four color components (default 0 for forth component)!"
+            )
+        else:
+            self._bpcs = bpcs
```

Comment:
I am not sure I understand what you're trying to do here. If `0` is supposed to be the default for the fourth component shouldn't you allow passing both 3 and 4 values and act accordingly?

Reply:
I addressed that in cf250635da3339dd7bf5452e54a41ded90cc0fdf.
I'm also not sure what I wanted to do here.
Thanks!